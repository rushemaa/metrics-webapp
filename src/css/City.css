.city-map {
  max-width: 50px;
}
