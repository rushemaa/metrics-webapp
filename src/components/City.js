import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import ArrowCircleRightOutlinedIcon from '@mui/icons-material/ArrowCircleRightOutlined';
import '../css/City.css';

export default function City({ city, breweries }) {
  return (
    <li className="city">
      <Link to="details" state={{ city, breweries }}>
        <div className="arrowRigght">
          <ArrowCircleRightOutlinedIcon />
        </div>
        <div>
          <h3>{city}</h3>
          <h5>{breweries}</h5>
        </div>
      </Link>
    </li>
  );
}

City.defaultProps = {
  city: '',
  breweries: 0,
};

City.propTypes = {
  city: PropTypes.string,
  breweries: PropTypes.number,
};
