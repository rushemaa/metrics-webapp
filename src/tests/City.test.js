import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import City from '../components/City';

describe('Test for render city details', () => {
  it('snapshot for city details', () => {
    const mockProp = {
      city: 'Austin',
      breweries: 26,
    };
    const { asFragment } = render(
      <BrowserRouter>
        <City prop={mockProp} />
      </BrowserRouter>,
    );
    expect(asFragment(<City prop={mockProp} />)).toMatchSnapshot();
  });
});
