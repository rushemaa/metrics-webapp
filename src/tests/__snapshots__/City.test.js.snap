// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for render city details snapshot for city details 1`] = `
<DocumentFragment>
  <li
    class="city"
  >
    <a
      href="/details"
    >
      <div
        class="arrowRigght"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowCircleRightOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
          />
        </svg>
      </div>
      <div>
        <h3 />
        <h5>
          0
        </h5>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
